nomos-services/cryptarchia-consensus/src/lib.rs

```diff
@@ -553,29 +695,51 @@ where
         blob_selector: BS,
         cl_mempool_relay: MempoolRelay<ClPool::Item, ClPool::Item, ClPool::Key>,
         da_mempool_relay: MempoolRelay<DaPoolAdapter::Payload, DaPool::Item, DaPool::Key>,
+        sampling_relay: SamplingRelay<SamplingBackend::BlobId>,
     ) -> Option<Block<ClPool::Item, DaPool::Item>> {
         let mut output = None;
         let cl_txs = get_mempool_contents(cl_mempool_relay);
         let da_certs = get_mempool_contents(da_mempool_relay);
-
-        match futures::join!(cl_txs, da_certs) {
-            (Ok(cl_txs), Ok(da_certs)) => {
+        let blobs_ids = get_sampled_blobs(sampling_relay);
+        match futures::join!(cl_txs, da_certs, blobs_ids) {
+            (Ok(cl_txs), Ok(da_blobs_info), Ok(blobs_ids)) => {
                 let Ok(block) = BlockBuilder::new(tx_selector, blob_selector)
                     .with_cryptarchia_builder(Builder::new(parent, proof))
                     .with_transactions(cl_txs)
-                    .with_blobs_certificates(da_certs)
+                    .with_blobs_info(
+                        da_blobs_info.filter(move |info| blobs_ids.contains(&info.blob_id())),
+                    )
                     .build()
                 else {
                     panic!("Proposal block should always succeed to be built")
                 };
                 tracing::debug!("proposed block with id {:?}", block.header().id());
                 output = Some(block);
             }
-            (Err(_), _) => tracing::error!("Could not fetch block cl transactions"),
-            (_, Err(_)) => tracing::error!("Could not fetch block da certificates"),
+            (tx_error, da_certificate_error, blobs_error) => {
+                if let Err(_tx_error) = tx_error {
+                    tracing::error!("Could not fetch block cl transactions");
+                }
+                if let Err(_da_certificate_error) = da_certificate_error {
+                    tracing::error!("Could not fetch block da certificates");
+                }
+                if let Err(_blobs_error) = blobs_error {
+                    tracing::error!("Could not fetch block da blobs");
+                }
+            }
         }
         output
     }
+
+    fn validate_block(
+        block: &Block<ClPool::Item, DaPool::Item>,
+        sampled_blobs_ids: &BTreeSet<DaPool::Key>,
+    ) -> bool {
+        let validated_blobs = block
+            .blobs()
+            .all(|blob| sampled_blobs_ids.contains(&blob.blob_id()));
+        validated_blobs
+    }
 }
```

Comment:
This is very temporary and will have to include many other things later one. But I think for now will do.